### DISCUSSION TITLE
Offline Sessions and Step Up Authentication

### DISCUSSION THREAD

**Discussion by dominikschlosser** [keycloak/keycloak#28515] [Q&A]
If you request offline_access, you get an Offline Token instead of a Refresh Token. If you use this Offline Token to get a new Access Token (refresh_token grant type), you get an Access Token based on the Offline Session.
As long as you only have a single Authentication this doesnt really matter, since the Offline Session is copied from the Online Session after the Flow has successfully finished. But if you do Step Up-Authentication afterwards and maybe change some session notes as a result of it, you will never see them in your Access Tokens (unless you write a custom Token Mapper which looks up the corresponding Online Session somehow).

Is this a supported use case? How should we deal with this?

**Comment by sschu** [comment]
In my opinion, offline tokens should really just be that - offline tokens. They should be completely separate from the (online) session that created them. So you can just go ahead and refresh them, but things like step-up don't work - you need to start a new authcode flow for that.
